### PR TITLE
data_platform_libs update

### PR DESCRIPTION
**Pull request by juditnovak**
## Issue

Some tests were executed with incorrect mocks. Thus we had "fake" green results. The latest `data-interfaces` changes highlight on that.

Also some tests (suffering of the above) were irrelevant for PGbouncer and could be removed.

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -660,56 +676,3 @@ def test_delete_existing_password_secrets(self, _):
                 "Non-existing field 'non-existing-secret' was attempted to be removed"
                 in self._caplog.text
             )
-
-    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    def test_migration_from_databag(self, scope, is_leader, _):
```
Comment (juditnovak) [inline tests/unit/test_charm.py:667]: This test is relevant when secrets support with old names was available in the past. I.e. we may have a rolling upgrade, where we still keep secrets in the databag (using old names) until the first time they are updated.

As I checked the PGbouncer code, we're not doing that here.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -660,56 +676,3 @@ def test_delete_existing_password_secrets(self, _):
...
-        assert self.harness.charm.model.get_secret(label=f"pgbouncer.{scope}")
-        assert self.harness.charm.get_secret(scope, "operator-password") == "blablabla"
-        assert "operator-password" not in self.harness.get_relation_data(
-            self.rel_id, getattr(self.charm, scope).name
-        )
-
-    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    def test_migration_from_single_secret(self, scope, is_leader, _):
```
Comment (juditnovak) [inline tests/unit/test_charm.py:689]: This test was checking for a transitional phase between secrets being registered by URL and from secrets being tracked using labels. PGbouncer should have never had this phase, thus the test is irrelevant.

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -617,35 +572,96 @@ def test_delete_password(self):
...
     @patch_network_get(private_address="1.1.1.1")
     @pytest.mark.usefixtures("use_caplog")
     def test_delete_existing_password_secrets(self, _):
         """NOTE: currently ops.testing seems to allow for non-leader to remove secrets too!"""
         with self.harness.hooks_disabled():
             self.harness.set_leader(True)
-        self.harness.charm.set_secret("app", "operator-password", "somepw")
-        self.harness.charm.remove_secret("app", "operator-password")
-        assert self.harness.charm.get_secret("app", "operator-password") is None
+        self.harness.charm.set_secret("app", "monitoring-password", "somepw")
```
Comment (juditnovak) [inline tests/unit/test_charm.py:643]: Changed to `monitoring-user`, as that's the secret field declared for the charm.

File: tests/unit/test_charm.py [hunk 4]
```diff
@@ -617,35 +572,97 @@ def test_delete_password(self):
...
+        self.harness.charm.remove_secret("app", "monitoring-password")
+        assert self.harness.charm.get_secret("app", "monitoring-password") is None
 
         with self.harness.hooks_disabled():
             self.harness.set_leader(False)
-        self.harness.charm.set_secret("unit", "operator-password", "somesecret")
-        self.harness.charm.remove_secret("unit", "operator-password")
-        assert self.harness.charm.get_secret("unit", "operator-password") is None
+        self.harness.charm.set_secret("unit", "csr", "somesecret")
```
Comment (juditnovak) [inline tests/unit/test_charm.py:650]: Using `csr`, as it is a valid secret for the charm on the `unit` level

**Review by dragomirp — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]

**Review by dragomirp — APPROVED** [review]

**Pull request merged**